Monika After Story/game/script-apologies.rpy

```diff
@@ -0,0 +1,250 @@
+#Create an apology db for storing our times
+default persistent._mas_apology_time_db = {}
+
+#Create a generic apology db. We'll want to know how many times the player has apologized for mas_apology_reason
+#Allows us the ability to apply diminishing returns on affection for repeated use of the same apology
+default persistent._mas_apology_reason_use_db = {}
+
+init -10 python in mas_apology:
+    apology_db = {}
+
+
+#going to need this post ev_handler init
+init 20 python:
+    def mas_checkApologies():
+        #Let's not do extra work
+        if persistent._mas_apology_time_db == {}:
+            return
+
+        #Calculate the current total playtime to compare...
+        current_total_playtime = persistent.sessions['total_playtime'] + (datetime.datetime.now() - persistent.sessions['current_session_start'])
+
+        #Iter thru the stuffs in the apology time tb
+        for ev, td in persistent._mas_apology_time_db.items():
+            if current_total_playtime >= td[0] or datetime.date.today() >= td[1]:
+                #Pop the ev_label from the time db and lock the event label. You just lost your chance
+                store.mas_hideEVL(ev,'APL',lock=True)
+                del persistent._mas_apology_time_db[ev]
```

Comment:
you should just loop over the keys, then take + unpack the data for comparison, and then use the `pop` function for the dict to take out the data.

to get a list copy of keys:
`some_dict.keys()`

to remove data from dict:
`some_dict.pop(key)`

also, use `mas_lockEVL`